Clinical-Genomics/scout
# Matchmaker integration. Complete

This PR contains the entire code that adds/updates or removes patients from patientMatcher, sends internal match requests to match scout patients or requests to external nodes. It has also an interface to visualize the submitted data  internal and external matches results.

Instructions to test this PR:

1. Install patientMatcher(https://github.com/Clinical-Genomics/patientMatcher). There's no need to create username and password for its database for testing purposes, so skip Prerequisites.

2. Using the patientMatcher cli, add a client for scout as described here: https://github.com/Clinical-Genomics/patientMatcher#adding-a-client-to-the-database. For instance you could use this command:
```bash
pmatcher add client -id scout_test -token test_token -url scout_test_url
```

3. Run patientMatcher on a different port than the one you use to run scout. For instance:
```bash
pmatcher run -p 9020
```

4. Make sure that you don't see anything matchmaker-related in the case page of scout.

5. Add a 'mme_submitter' role to your scout test user:
![image](https://user-images.githubusercontent.com/28093618/52401714-aa423b00-2ac2-11e9-9313-473dd09e336d.png)

6. Edit the scout config file(in server/config.py) adding these params:
#MatchMaker related parameters
MME_ACCEPTS = 'application/vnd.ga4gh.matchmaker.v1.0+json'
MME_URL = 'http://localhost:9020'
MME_TOKEN = 'test_token' #this should be the same of when you created the client in step 2

7. Go to the case page and see that now you can submit max 3 pinned variants to matchmaker (genes only or complete variants). You can optionally submit patient sex, hpo terms and OMIM diagnoses. The form should look like this:
![image](https://user-images.githubusercontent.com/28093618/52900867-05c7a380-31fc-11e9-849a-455caf32634b.png)

8. Once you submit the patient you can update your submission by changing pinned variants, or the other options described above. You should also be able to delete the patient from matchmaker. The form should look like this:
![image](https://user-images.githubusercontent.com/28093618/52900878-39a2c900-31fc-11e9-8b14-a9c96e5a71b4.png)

9. From the same interface you will be able to run matches against scout patients in MatchMaker:
![image](https://user-images.githubusercontent.com/28093618/52901008-c306cb00-31fd-11e9-83a4-329412165840.png)
When testing on stage we will be connected to an external node and the option will be showed here.

10. Once having runned one of more match request against the internal node check that there are results in the matches interface:
![image](https://user-images.githubusercontent.com/28093618/52901044-3d374f80-31fe-11e9-9e02-26143865ddb2.png)
